### PR TITLE
parse HTML-style and new <%!-- --%> tags as comments

### PR DESCRIPTION
**Pull request by the-mikedavis**
connects https://github.com/connorlay/tree-sitter-eex/issues/3
see also https://github.com/elixir-lang/elixir/pull/11505

Plus I threw in the HTML `<!-- .. -->` style comments too, even though I suppose if you're using HEEx there isn't much reason to ever use an old-fashioned HTML comment :upside_down_face:

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -155,8 +155,30 @@ module.exports = grammar({
...
+
+    _bang_comment: $ => seq(
+      '<%!--',
+      prec.left(
+        seq(
+          repeat1(/[^-]+|-/),
+          '--%>'
+        )
+      )
+    ),
```
Comment (connorlay) [inline grammar.js:179]: Do you know if this works with HEEx today, or is it going to be added in the future?

**Review by connorlay — APPROVED** [review]
Looks great to me, thanks!

File: grammar.js (hunk 1 above)
Comment (the-mikedavis) [reply grammar.js:179]: I believe this does not currently work.

It isn't in the changelog yet but I think `<%!-- --%>` style arrives for EEx in 1.14 and I see that HEEx works as an "[engine](https://github.com/phoenixframework/phoenix_live_view/blob/e5084d93221c1f11114f95d83d675b5ce4a51578/lib/phoenix_live_view/html_engine.ex#L19)" of EEx, so I assume HEEx gets all the EEx improvements (although I did not look into that engine behavior, I might be wrong.

Maybe we should hold off on this until it's announced?

Comment (connorlay) [reply grammar.js:179]: > Maybe we should hold off on this until it's announced?

I'd like to wait on `<%!-- -->` until we know it is going to be supported in HEEx. Perhaps @josevalim knows the answer?

Comment (josevalim) [reply grammar.js:179]: `<%— —%>` will automatically work on EEx once v1.14 is out!

Comment (josevalim) [reply grammar.js:179]: Errr, both HEEx and EEx.

Comment (connorlay) [reply grammar.js:179]: Awesome, thanks for confirming!

**Pull request merged**